blobfuse/directoryapis.cpp

```diff
@@ -142,6 +159,46 @@ int azs_readdir(const char *path, void *buf, fuse_fill_dir_t filler, off_t, stru
                         stbuf.st_nlink = 1;
                         stbuf.st_size = listResults[i].content_length;
                         fillerResult = filler(buf, prev_token_str.c_str(), &stbuf, 0); // TODO: Add stat information.  Consider FUSE_FILL_DIR_PLUS.
+
+                        /// List attribute cache: cache the found file locally without the contents
+                        if(list_attribute_cache == true)
+                        {
+                            off_t length = listResults[i].content_length;
+
+                            // mutex lock the file path
+                            auto fmutex = file_lock_map::get_instance()->get_mutex(("/" + listResults[i].name).c_str());
```

Comment:
are you locking correct key? is it "a path as seen by user app"?

Reply:
Yes - I just verified. Both in azs_open, and in here, the path is exactly same. It is the relative path under the mountpoint